src/fheroes2/campaign/campaign_data.cpp

```diff
@@ -800,38 +810,41 @@ namespace Campaign
                 description += monsters.back().GetMultiName();
             }
 
-            description += _( " will be willing to always join your army." );
+            description += _( " will be willing to join your army." );
             return description;
         }
         case CampaignAwardData::TYPE_GET_ARTIFACT: {
             std::string description( _( "\"%{artifact}\" artifact will be carried over the scenario." ) );
-            StringReplace( description, "%{artifact}", Artifact( static_cast<int>( _subType ) ).GetName() );
+            StringReplace( description, "%{artifact}", Artifact( _subType ).GetName() );
             return description;
         }
-        case CampaignAwardData::TYPE_CARRY_OVER_FORCES:
+        case CampaignAwardData::TYPE_CARRY_OVER_FORCES: {
             return _( "The army will be carried over the scenario." );
+        }
         case CampaignAwardData::TYPE_RESOURCE_BONUS: {
-            std::string description( _( "The kingdom will have additional %{resource} at the start of the scenario." ) );
-            StringReplace( description, "%{resource}", Resource::String( static_cast<int>( _subType ) ) );
+            std::string description( _( "The kingdom will have +%{count} %{resource} each day." ) );
+            StringReplace( description, "%{count}", std::to_string( _amount ) );
+            StringReplace( description, "%{resource}", Resource::String( _subType ) );
             return description;
         }
         case CampaignAwardData::TYPE_GET_SPELL: {
             std::string description( _( "\"%{spell}\" spell will be carried over the scenario." ) );
-            StringReplace( description, "%{spell}", Spell( static_cast<int>( _subType ) ).GetName() );
+            StringReplace( description, "%{spell}", Spell( _subType ).GetName() );
             return description;
         }
         case CampaignAwardData::TYPE_HIREABLE_HERO: {
-            std::string description( _( "%{hero} hero can be hired in the scenario." ) );
-            StringReplace( description, "%{hero}", Heroes( static_cast<int>( _subType ), 0 ).GetName() );
+            std::string description( _( "%{hero} can be hired in the scenario." ) );
+            StringReplace( description, "%{hero}", Heroes( _subType, 0 ).GetName() );
             return description;
         }
         case CampaignAwardData::TYPE_DEFEAT_ENEMY_HERO: {
-            std::string description( _( "Defeated %{hero} hero will not appear in the subsequent scenarios." ) );
-            StringReplace( description, "%{hero}", Heroes( static_cast<int>( _subType ), 0 ).GetName() );
+            std::string description( _( "Defeated %{hero} will not appear in the subsequent scenarios." ) );
```

Comment:
Ex. "Lord Corlagon has been defeated and will not appear in the subsequent scenarios."
This award only appears once you've defeated that hero in a scenario in a campaign.
```suggestion
            std::string description( _( "%{hero} has been defeated and will not appear in the subsequent scenarios." ) );
```